pkg/apis/policy/v1alpha1/clusterimagepolicy_validation.go

```diff
@@ -240,6 +245,12 @@ func (identity *Identity) Validate(ctx context.Context) *apis.FieldError {
 	if identity.SubjectRegExp != "" {
 		errs = errs.Also(ValidateRegex(identity.SubjectRegExp).ViaField("subjectRegExp"))
 	}
+	if identity.SubjectRegExp == "" && identity.Subject == "" {
+		errs = errs.Also(apis.ErrMissingField("subject", "subjectRegExp").At(apis.WarningLevel))
+	}
+	if identity.IssuerRegExp == "" && identity.Issuer == "" {
+		errs = errs.Also(apis.ErrMissingField("issuer", "issuerRegExp").At(apis.WarningLevel))
+	}
```

Comment:
I think that we may actually want these to be errors.  If we see an identity without one of these, then it is an error.

I think what should be a warning is an empty list of Identities: https://github.com/sigstore/policy-controller/blob/6a0a54404d9c341e70faf98fd1b0f7293efd64e4/pkg/apis/policy/v1alpha1/clusterimagepolicy_validation.go#L175-L177

I don't see that check.

Reply:
Well, we did not require either to be there. So having subject, but no issuer was valid choice so I don't think they should be errors.
I think having empty identities as a warning makes sense for sure though.

Reply:
For these, do you want to indicate to folks that these will become required in a future release as part of the error message?

Reply:
Yeah, but I'll do that as part of the work below for proper e2e tests that weren't there. Want to figure out when we start enforcing it :)